### ISSUE TITLE
piping into ffmpeg

### ISSUE THREAD

**Issue by AlaShiban** [Genymobile/scrcpy#768]
I may be missing something very obvious (apologies!) - but how do I pipe the output of scrcpy to FFmpeg for further processing/branching on windows? An example of using a b/w filter would be great. Thanks! (for simplicity)

**Comment by AlaShiban** [comment]
Hey @rom1v, any insights?

**Comment by rom1v** [comment]
The video stream received from the device contains the raw H.264 stream "interlaced with" some headers (containing timestamps and which delimit packet boundaries, see [v1.10 release notes](https://github.com/Genymobile/scrcpy/releases/tag/v1.10)).

Therefore, it cannot be exposed "as is" (this is a "private" protocol, nothing will be able to read it).
Streaming the raw H.264 stream would not be sufficient either, since timestamps would be missing (which is fine for live streaming locally, not for post-processing or recording).

So basically, this would require to expose a new "stream" feature which exposes a video stream in a specific format (live mkv? rtp?) in some way (socket? file? stdout?). This is not possible for now.

**Comment by rom1v** [comment]
(In the previous message, I answered to "piping into ffmeg", understood as the command-line ffmpeg)

Since scrcpy uses ffmpeg (libav) to decode frames, it would be possible to add code to do whatever filter you want, but there are no "module" or "plugins", so you need to fork it if you want to do this.

**Comment by AlaShiban** [comment]
Thanks!
